src/ai/backend/manager/models/domain.py

```diff
@@ -282,3 +284,27 @@ async def delete_kernels(
         if result.rowcount > 0:
             log.info('deleted {0} domain\'s kernels ({1})', result.rowcount, domain_name)
         return result.rowcount
+
+    @classmethod
+    async def domain_has_active_kernels(
+        cls,
+        conn: SAConnection,
+        domain_name: str,
+    ) -> bool:
+        """
+        Check if the domain does not have active kernels.
+
+        :param conn: DB connection
+        :param domain_name: domain's name
+
+        :return: True if the domain has some active kernels.
+        """
+        from . import kernels, AGENT_RESOURCE_OCCUPYING_KERNEL_STATUSES
+        query = (
+            sa.select([sa.func.count()])
+            .select_from(kernels)
+            .where((kernels.c.domain_name == domain_name) &
+                   (kernels.c.status.in_(AGENT_RESOURCE_OCCUPYING_KERNEL_STATUSES)))
+        )
+        active_kernel_count = await conn.scalar(query)
+        return True if active_kernel_count > 0 else False
```

Comment:
This may look more explicit, but I think it's better to just rely on the language semantics:
```suggestion
        return (active_kernel_count > 0)
```